server/stream.go

```diff
@@ -779,12 +784,8 @@ func (mset *Stream) setupStore(fsCfg *FileStoreConfig) error {
 func (mset *Stream) storeUpdates(md, bd int64, seq uint64, subj string) {
 	// If we have a single negative update then we will process our consumers for stream pending.
 	// Purge and Store handled separately inside individual calls.
-	if md == -1 {
-		mset.mu.RLock()
-		for _, o := range mset.consumers {
-			o.decStreamPending(seq, subj)
-		}
-		mset.mu.RUnlock()
+	if md == -1 && seq > 0 {
+		mset.decq <- &decp{seq, subj}
```

Comment:
Yeah, but you now have the problem that this channel may be full and will block! And if this gets called under mset lock, then the "internalDecLoop" may be blocked wanted to get the RLock()..

Reply:
Yes I thought about that.. Unfortunately this has folks blocked, main branch is broke for any stream with limits.
Let me try again since I have a fresher set of eyes.

Reply:
I understand. In that case, the version with atomic (although wrong) is less dangerous. The worst case is that the map would be iterated without locking - which a test may report as data race - but will not deadlock like this approach could.

Reply:
Sorry, no acc will not change either.

Reply:
Agree deadlock is what needs to be first priority, then the preconditions, etc.